test/powershell/Modules/Microsoft.PowerShell.Utility/Json.Tests.ps1

```diff
@@ -345,6 +345,258 @@ Describe "Json Tests" -Tags "Feature" {
             ValidateSampleObject -result $result -hasEmbeddedSampleObject
         }
 
+		It "ConvertFrom-Json correctly processes datetimes vs strings using newtonsoft conventions" {
+			# Create all the various "standard" date formats from a source date 2008-09-22T14:01:54.9571247Z
+			# Then we deserialize them and verify which ones newtonsoft decides to make into datetimes and which ones newtonsoft keeps as strings
+			# For the ones that are datetimes, make sure they retain all of the various formats correctly
+			# for the ones that are strings just verify that they are actually strings.
+
+			$json = @"
+{
+	"date-s-should-parse-as-datetime": "2008-09-22T14:01:54",
+	"date-upperO-should-parse-as-datetime": "2008-09-22T14:01:54.9571247Z",
+	
+	"date-o-should-parse-as-string": "2019-12-17T06:14:06 +06:00",
+	"date-upperD-should-parse-as-string": "Monday, September 22, 2008",
+	"date-f-should-parse-as-string": "Monday, September 22, 2008 2:01 PM",
+	"date-upperF-should-parse-as-string": "Monday, September 22, 2008 2:01:54 PM",
+	"date-g-should-parse-as-string": "9/22/2008 2:01 PM",
+	"date-upperG-should-parse-as-string": "9/22/2008 2:01:54 PM",
+	"date-m-should-parse-as-string": "September 22",
+	"date-upperM-should-parse-as-string": "September 22",
+	"date-upperR-should-parse-as-string": "Mon, 22 Sep 2008 14:01:54 GMT",
+	"date-t-should-parse-as-string": "2:01 PM",
+	"date-upperT-should-parse-as-string": "2:01:54 PM",
+	"date-u-should-parse-as-string": "2008-09-22 14:01:54Z",
+	"date-upperU-should-parse-as-string": "Monday, September 22, 2008 2:01:54 PM",
+	"date-upperY-should-parse-as-string": "September 2008",
+	"date-y-should-parse-as-string": "September 2008"
+}
+"@
+			foreach ($asHashtableOption in @($True, $false)) {
```

Comment:
```suggestion
			foreach ($asHashtableOption in @($True, $False)) {
```

Reply:
Ive incorporated this change into e9eb1b9